### PR TITLE
[e2e][parallel] Allow set custom config on Parallel

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba this seems a reason why downgrade scoper build can't work via Parallel, it because it can't read config via `--c` because the script is target with `--config`:

https://github.com/rectorphp/rector-src/blob/f900476469c411ad0ae25e1cb32637b392d72d22/build/downgrade-rector.sh#L26

This PR apply:

**1.** updated `WorkerCommandLineFactory` to append config for it based on option 'config' https://github.com/rectorphp/rector-src/pull/1620/files#diff-1c7bf20096eedf4dac69e487b7d1e3ad3f9cc122b98d54e145c2f463a2facab9

```php
        if ($input->hasOption(Option::CONFIG)) {
            $workerCommandArray[] = '--config';
            $workerCommandArray[] = $input->getOption(Option::CONFIG);
        }
```

If no `--config` input, it will got config from `rector.php`. 

**2.** Add `e2e` tests for its demo with pass custom config into input.

### CODE REVIEW

**Comment by samsonasik** [comment]
It seems `-c` with '--config` got different result:

```php
$input->hasOption(Option::CONFIG)
```

return true on `-c` passed, while not for `--config`

**Comment by samsonasik** [comment]
I updated e2e script to test passed `--config` as well

**Comment by samsonasik** [comment]
Fixed 🎉

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

Next PR is to apply `Option::PARALLEL` to downgrade config.

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -45,7 +46,15 @@ jobs:
...
+                working-directory: ${{ matrix.directory }}
+                if: ${{ matrix.directory == 'e2e/parallel-custom-config' }}
+
+            -   run: php ../e2eTestRunner.php --config custom/config/rector.php
+                working-directory: ${{ matrix.directory }}
+                if: ${{ matrix.directory == 'e2e/parallel-custom-config' }}
+
+            -   run: php ../e2eTestRunner.php
                 working-directory: ${{ matrix.directory }}
+                if: ${{ matrix.directory != 'e2e/parallel-custom-config' }}
```
Comment (TomasVotruba) [inline .github/workflows/e2e.yaml:60]: We could put the "if" right into the matrix:

```yaml
   matrix:
                php_version: ['8.1']
                actions:
                     -
                          directory: e2e/template-extends
                          runner: php ../e2eTestRunner.php -c custom/config/rector.php
```

Comment (TomasVotruba) [reply .github/workflows/e2e.yaml:60]: I'm merging this PR to keep original fix and this syntax improvement separated :+1:

**Comment by TomasVotruba** [comment]
Thank you :+1:

**Comment by samsonasik** [comment]
@TomasVotruba here the original PR why `-c` vs `--config` is needed in e2e tests parallel

**Pull request merged**